services/repository/files/update.go

```diff
@@ -539,11 +540,20 @@ func CreateOrUpdateFile(ctx context.Context, t *TemporaryUploadRepository, file
 			return err
 		}
 		if !exist {
-			_, err := file.ContentReader.Seek(0, io.SeekStart)
-			if err != nil {
-				return err
+			var lfsContentReader io.Reader
+			if file.Operation != "rename" {
+				if _, err := file.ContentReader.Seek(0, io.SeekStart); err != nil {
+					return err
+				}
+				lfsContentReader = file.ContentReader
+			} else {
+				if lfsContentReader, err = oldEntry.Blob().DataAsync(); err != nil {
```

Comment:
This is the pointer content not the lfs file content?

Reply:
It's the new lfs file content. `file.ContentReader` is the text in the web editor on a create or update operation; `oldEntry.Blob().DataAsync()` is the data from the old tree path which was stored in git normally without lfs, this is assured by previous operations, `renameFileHash` only returns an lfs pointer that is `!= nil` if the content at the old tree path is not an lfs pointer

Reply:
For a rename operation, it will only move the lfs pointer file from a tree path to another. It seems no content needs to be stored into LFS store.

Reply:
It's possible that the user moves something from "normal" git to lfs

Reply:
But why stored the `oldentry` but not the `newentry` if it's a rename from git data to a lfs data?

Reply:
The purpose of `oldEntry` is just to get the git data of the old path, either to read it as lfs pointer and moving the lfs data to git if the old path is in lfs but the new isn't, or to read it as normal data and then moving the data to lfs if the new path is in lfs but the old isn't - there is no need for a newentry